src/mlpack/tests/nmf_test.cpp

```diff
@@ -222,4 +222,65 @@ BOOST_AUTO_TEST_CASE(SparseNMFALSTest)
       1e-5);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
+/**
+ * Check if all elements in W and H are non-negative.
+ * Default Case.
+ * Random Acol initilization and distance minimization update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFDefaultTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  AMF<> nmf;
+  nmf.Apply(v, r, w, h);
+
+  BOOST_REQUIRE(arma::all(vectorise(std::move(w)) >= 0)
```

Comment:
No need to use ``move`` here, this should make the line shorter and more readable.